apps/question-bank/src/test/java/gov/cdc/nbs/questionbank/page/content/subsection/SubsectionGrouperTest.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    @Test
+    void should_not_update_null_columWith() {
+        GroupSubSectionRequest request = new GroupSubSectionRequest(100l, "BLOCK_X", getUnValidBatchColumnWidth());
+        Long userId = 456L;
+        UpdateSubSectionException exception = assertThrows(UpdateSubSectionException.class, () -> grouper.group(1l, request, userId));
+        assertEquals("batchTableColumnWidth is required", exception.getMessage());
+    }
+
+    @Test
+    void should_not_update_columnWith_lessThan_100() {
+        GroupSubSectionRequest request = new GroupSubSectionRequest(100l, "BLOCK_X", getUnValidBatchColumnWidthTotal());
+        Long userId = 456L;
+        UpdateSubSectionException exception = assertThrows(UpdateSubSectionException.class, () -> grouper.group(1l, request, userId));
+        assertEquals("the total of batchTableColumnWidth must calculate to 100", exception.getMessage());
+    }
+
+    List<GroupSubSectionRequest.Batch> getValidBatchList() {
+        List<GroupSubSectionRequest.Batch> batchList = new ArrayList<>();
+        batchList.add(new GroupSubSectionRequest.Batch(101l, 'Y', "header1", 25));
+        batchList.add(new GroupSubSectionRequest.Batch(102l, 'Y', "header2", 75));
+        return batchList;
+    }
+
+    List<GroupSubSectionRequest.Batch> getUnValidBatchColumnWidth() {
+        List<GroupSubSectionRequest.Batch> batchList = new ArrayList<>();
+        batchList.add(new GroupSubSectionRequest.Batch(101l, 'Y', "header1", 0));
+        batchList.add(new GroupSubSectionRequest.Batch(102l, 'Y', "header2", 75));
+        return batchList;
+    }
+
+    List<GroupSubSectionRequest.Batch> getUnValidBatchColumnWidthTotal() {
+        List<GroupSubSectionRequest.Batch> batchList = new ArrayList<>();
+        batchList.add(new GroupSubSectionRequest.Batch(101l, 'Y', "header1", 30));
+        batchList.add(new GroupSubSectionRequest.Batch(102l, 'Y', "header2", 40));
+        return batchList;
+    }
+
+
+    @Test
+    void should_unGroup_subsection() {
+        WaTemplate page = mock(WaTemplate.class);
+        when(entityManager.find(WaTemplate.class, 1l)).thenReturn(page);
+        doNothing().when(entityManager).flush();
+        WaUiMetadata waUiMetadata = new WaUiMetadata();
+        waUiMetadata.setId(100l);
+        List<Long> batches = new ArrayList<>(Arrays.asList(101L, 102L));
+        UnGroupSubSectionRequest request = new UnGroupSubSectionRequest(100l, batches);
+        when(page.unGroupSubSection(any())).thenReturn(waUiMetadata);
+        Long userId = 456L;
+        ResponseEntity<String> result = grouper.unGroup(1l, request, userId);
+        assertEquals("Subsection " + waUiMetadata.getId() + " is UnGrouped Successfully ", result.getBody());
+    }
+
+    @Test
+    void testGroupSubSection() {
+        // Create a mock for WaUiMetadata
+        Long subSectionId = 1L;
+        WaUiMetadata subsection = mock(WaUiMetadata.class);
+        when(subsection.getId()).thenReturn(subSectionId);
```

Comment:
nit:  Not sure if there is a need for all of the mocking, these objects aren't very large